src/components/Popup/Popup.js

```diff
@@ -21,46 +25,114 @@ const defaultProps = {
   popupComponents: null,
 };
 
-const Popup = ({
-  clickedFeatureInfo,
-  map,
-  popupComponents,
-  dispatchSetClickedFeatureInfo,
-}) => {
-  if (!clickedFeatureInfo) {
-    return null;
+class Popup extends Component {
+  constructor(props) {
+    super(props);
+    const { clickedFeatureInfo } = this.props;
+
+    this.state = {
+      featIndex: 0,
+      featuresList: clickedFeatureInfo ? clickedFeatureInfo.features : [],
```

Comment:
Why do we need this state variable? We already have `clickedFeatureInfo.feature` which updates when necessary.